traveler_core_kit/src/main/java/com/guestlogix/travelercorekit/utilities/TravelerPrefs.java

```diff
@@ -0,0 +1,42 @@
+package com.guestlogix.travelercorekit.utilities;
+
+import android.content.Context;
+import android.content.SharedPreferences;
+
+public class TravelerPrefs {
+
+    private static final String TRAVELER_PREFS = "TRAVELER_PREFS";
+    public static final String TRAVELER_SDK_ENDPOINT = "TRAVELER_SDK_ENDPOINT";
+    public static final String TRAVELER_AUTH_ENDPOINT = "TRAVELER_AUTH_ENDPOINT";
+
+    private static volatile TravelerPrefs travelerPrefs;
+    private static SharedPreferences sharedPreferences;
+
+    private TravelerPrefs(Context context){
+        sharedPreferences = context.getSharedPreferences(TRAVELER_PREFS, Context.MODE_PRIVATE);
+    }
+
+    public static TravelerPrefs getInstance(Context context){
+        if (travelerPrefs == null){
+            travelerPrefs = new TravelerPrefs(context);
+        }
+
+        return travelerPrefs;
+    }
+
+    public void save(String key, String value) {
+        editor().putString(key, value).apply();
+    }
+
+    public void remove(String key) {
+        editor().remove(key).apply();
+    }
+
+    public String get(String key, String defaultValue) {
+        return sharedPreferences.contains(key) ? sharedPreferences.getString(key, defaultValue) : defaultValue;
+    }
+
+    private SharedPreferences.Editor editor() {
+        return sharedPreferences.edit();
```

Comment:
This pattern makes it confusing..just do all these in your remove and save methods

Reply:
done